src/mtcp/mtcp_restart.c

```diff
@@ -864,6 +865,10 @@ static int read_one_memory_area(int fd)
         imagefd = mtcp_sys_open (area.name, O_CREAT|O_RDWR, 0);
         if (imagefd >= 0) {
           mtcp_sys_unlink(area.name);
+        } else if (fix_parent_directories(area.name) == area.name) {
```

Comment:
Shouldn't we proactively create the directory tree, i.e. why not call fix_parent_directories() before we do mtcp_sys_open for the shared file? That way we can retain the `} else {` logic here.

With this patch, the `else` branch is dead code. `fix_parent_directories` will always return `area.name` or it will abort, in both cases the `else` branch is never taken.